support/k8s/k8s-workload-registrar/webhookhandler.go

```diff
@@ -38,25 +119,85 @@ func (h *WebhookHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 
-	var in admv1beta1.AdmissionReview
-	if err := json.NewDecoder(req.Body).Decode(&in); err != nil {
-		http.Error(w, "Malformed JSON body", http.StatusBadRequest)
+	var body []byte
+	if data, err := ioutil.ReadAll(req.Body); err == nil {
+		body = data
+	}
+
+	obj, gvk, err := deserializer.Decode(body, nil, nil)
+	if err != nil {
+		msg := fmt.Sprintf("Malformed JSON body: %v", err)
+		h.log.Error(msg)
+		http.Error(w, msg, http.StatusBadRequest)
 		return
 	}
 
-	if in.Request == nil {
-		http.Error(w, "AdmissionReview is missing request", http.StatusBadRequest)
+	log := h.log.WithFields(logrus.Fields{
+		"version": gvk.Version,
+		"kind":    gvk.Kind,
+	})
+	admit := newDelegateToV1AdmitHandler(h.controller.ReviewAdmission)
+	ctx := req.Context()
+
+	var responseObj runtime.Object
+	switch *gvk {
+	case admv1beta1.SchemeGroupVersion.WithKind("AdmissionReview"):
+		requestedAdmissionReview, ok := obj.(*admv1beta1.AdmissionReview)
+		if !ok {
+			msg := fmt.Sprintf("Expected v1beta1.AdmissionReview but got: %T", obj)
+			log.Error(msg)
+			http.Error(w, msg, http.StatusBadRequest)
+			return
+		}
+		responseAdmissionReview := &admv1beta1.AdmissionReview{}
+		responseAdmissionReview.SetGroupVersionKind(*gvk)
+		resp, err := admit.v1beta1(ctx, *requestedAdmissionReview)
+		if err != nil {
+			msg := fmt.Sprintf("internal error occurred: %v", err)
```

Comment:
nit: our log lines start with uppercase...
```suggestion
			msg := fmt.Sprintf("Internal error occurred: %v", err)
```